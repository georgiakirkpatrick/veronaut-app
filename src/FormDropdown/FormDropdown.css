.FormDropdown {
    width: 100%;
    margin-bottom: 2vh;
}

.FormDropdown > label {
    padding-bottom: 2vh;
}

.FormDropdown > p {
    font-weight: 400;
    font-size: 16px;
    padding: 5px 0;
    color: #585858
}

.FormDropdown > select {
    display: block;
    width: 100%;
    margin: 5px 0px 10px;
    padding: 10px;
    font: 400 16px/20px Poppins, sans-serif;
    border-radius: 5px;
    background: #fff;
    appearance: none;
}

.FormPromptWithSub {
    display: block;
    margin-bottom: 1vh;
    font: 400 16px/20px Poppins, sans-serif;
}

.FormPromptWithSub__prompt {
    padding-bottom: 2vh;
}

.FormPromptWithSub__subtitle {
    color: #585858
}