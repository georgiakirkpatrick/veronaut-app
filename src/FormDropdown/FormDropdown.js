import React from 'react'
import './FormDropdown.css'

const FormDropdown = props => {
    const {
        currentValue,
        id,
        handleChange,
        name,
        options,
        prompt,
        subprompt
    } = props

    return (
        <div key={id} className='FormDropdown'>
            <label htmlFor={id}>{prompt}</label>
            {subprompt && <p>{subprompt}</p>}
            <select
                required
                name={name}
                onChange={handleChange}
                value={currentValue}
            >
                {options.map((opt, index) => {
                    return <option key={index} id={opt.id} value={opt.value ? opt.value : opt.id}>{opt.text}</option>
                })}
            </select>
        </div>
    )
}

FormDropdown.defaultProps = {
    currentValue: '',
    id: 1,
    handleChange: () => {},
    name: '',
    options: [],
    prompt: '',
    subprompt: ''
}

export default FormDropdown