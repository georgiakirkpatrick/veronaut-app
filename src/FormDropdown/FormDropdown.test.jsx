import React from 'react'
import { render, screen } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import FormDropdown from './FormDropdown'

describe('FormDropdown', () => {
  it ('renders without crashing', () => {
    render(<FormDropdown />)
    screen.debug()
  })

  it ('render the UI as expected', () => {
    const tree = render(<FormDropdown />)
    expect(tree).toMatchSnapshot()
  })
})