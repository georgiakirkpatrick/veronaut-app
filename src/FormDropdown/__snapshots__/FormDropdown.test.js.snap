// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormDropdown render the UI as expected 1`] = `
<div
  className="FormDropdown"
>
  <label
    htmlFor={1}
  >
    
  </label>
  
  <select
    name=""
    onChange={[Function]}
    required={true}
    value=""
  />
</div>
`;
